server/engine.go

```diff
@@ -199,16 +199,28 @@ func (e *Engine) Find(query *Query, config *Config) (*FindResult, error) {
 	if atomicMaxUint32(&e.maxFindOpsInProgress, numFindOps) {
 		e.ErrorLog.Infof("Max simultaneous find ops in progress: %v", atomic.LoadUint32(&e.maxFindOpsInProgress))
 	}
+
+	// something is wrong if we go above 50 pending requests -> switch to detailed logging
+	if numFindOps > 50 {
```

Comment:
To make this thread safe, would I have to do something like this?
```
if atomic.CompareAndSwapUint32(&locker, 0, 1) {
  if numFindOps > 50 {
    e.ErrorLog.Level = log.Debug
  } else {
    e.ErrorLog.Level = log.Info
  }
}
defer atomic.StoreUint32(&locker, 0)
```
where `locker` is a global `uint32`

Reply:
It still wouldn't work, because one thread would come along, set it to Debug, then while that thread was busy processing it's request, another thread would come alone and set it back to Info

Reply:
All threads share the same Level, so fundamentally it just doesn't work.

Reply:
Sorry.. just *one more thing* - I would omit the "else" side of the branch, so that the default configured logging level still works.